### PR TITLE
Add Replace text with named capture groups example. fixes #243

### LINKED ISSUE #243: add "Replace text with named capture groups" example
This example should use [regex](https://crates.io/crates/regex) crate 
cc https://github.com/brson/rust-cookbook/issues/217

See [CONTRIBUTING.md](https://github.com/brson/rust-cookbook/blob/master/CONTRIBUTING.md) for details.

### PR DESCRIPTION
**Pull request by Phaiax**
#243 

I'm not sure about the category. (I put #242 into `basics`, but app development seems better now^^)

### CODE REVIEW

**Comment by Phaiax** [comment]
I stole this one from the regex crate

**Comment by budziq** [comment]
Hi @Phaiax thanks for all the submissions! 

Unfortunately I'll not be able to follow up with anything resembling review for the next two weeks due to holidays 😞 . I may drop some comments when Im on mobile later and I've asked other maintainers to drop from time to time and look into the PR's.

To answer your unspoken question, regex examples would go to basics section (atm the distinction between basics and app is quite arbitrary and this will change in the future)

File: src/app.md [hunk 1]
```diff
@@ -476,6 +477,35 @@ fn run() -> Result<()> {
...
+
+[ex-regex-replace-named]: #ex-regex-replace-named
+<a name="ex-regex-replace-named"></a>
+## Replace all occurrences of one text pattern with another pattern.
+
+[![regex-badge]][regex] [![cat-text-processing-badge]][cat-text-processing]
+
+Replaces all occurences of the hyphenated british english date pattern `2013-01-15` with its equivalent slashed american english date pattern `01/15/2013`.
+
+```
```
Comment (budziq) [inline src/app.md]: please add 
<pre> ```rust </pre>

otherwise the testrunner will not compile nor run it.

**Comment by brson** [comment]
I rebased and merged manually. Thanks @Phaiax

**Pull request closed without merging**